src/linker/pipeline.py

```diff
@@ -26,18 +24,30 @@ def run(self, runner: Callable, results_dir: Path) -> None:
                 results_dir=results_dir,
             )
 
-    def _validate(self, schema: "PipelineSchema") -> None:
-        for idx, implementation in enumerate(self.implementations):
-            # Check that all steps are accounted for and in the correct order
-            if implementation.step_name != schema.steps[idx].name:
-                return False
-            # Check that container exists
-            if (
-                not Path(f"{implementation._container_full_stem}.tar.gz").exists()
-                and not Path(f"{implementation._container_full_stem}.sif").exists()
-            ):
-                return False
-        return True
+    def _validate(self) -> None:
+        """Validates the pipeline against supported schemas."""
+
+        from linker.pipeline_schema import PIPELINE_SCHEMAS, PipelineSchema
+
+        def validate_pipeline(schema: PipelineSchema):
+            for idx, implementation in enumerate(self.implementations):
+                # Check that all steps are accounted for and in the correct order
+                if implementation.step_name != schema.steps[idx].name:
+                    return False
+                # Check that container exists
+                if (
+                    not Path(f"{implementation._container_full_stem}.tar.gz").exists()
+                    and not Path(f"{implementation._container_full_stem}.sif").exists()
+                ):
+                    return False
```

Comment:
We're missing the check that the implementation itself actually implements the step the user claims it implements (i.e. that the pipeline.yaml and implementation's metadata.yaml have the same step.

Reply:
Should be there now